internal/dag/gatewayapi_processor.go

```diff
@@ -150,15 +152,47 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 				if gatewayAllowMatches {
 					// Empty Selector matches all routes.
-					matchingRoutes = append(matchingRoutes, route)
+					matchingHTTPRoutes = append(matchingHTTPRoutes, route)
 				}
 			}
 		}
 
-		// Process all the routes that match this Gateway.
-		for _, matchingRoute := range matchingRoutes {
+		for _, route := range p.source.tlsroutes {
+			// Filter the TLSRoutes that match the gateway which Contour is configured to watch.
+			// RouteBindingSelector defines a schema for associating routes with the Gateway.
+			// If Namespaces and Selector are defined, only routes matching both selectors are associated with the Gateway.
+
+			// ## RouteBindingSelector ##
+			//
+			// Selector specifies a set of route labels used for selecting routes to associate
+			// with the Gateway. If this Selector is defined, only routes matching the Selector
+			// are associated with the Gateway. An empty Selector matches all routes.
+
+			nsMatches, err := p.namespaceMatches(listener.Routes.Namespaces, route.Namespace)
+			if err != nil {
+				p.Errorf("error validating namespaces against Listener.Routes.Namespaces: %s", err)
+			}
+
+			selMatches, err := selectorMatches(listener.Routes.Selector, route.Labels)
+			if err != nil {
+				p.Errorf("error validating routes against Listener.Routes.Selector: %s", err)
+			}
+
+			if selMatches && nsMatches {
```

Comment:
Is [this block of logic](https://github.com/projectcontour/contour/blob/e3cf805af778fe42ec39681dbf71b783b091007e/internal/dag/gatewayapi_processor.go#L138-L157) that we have for `HTTPRoutes` not needed for `TLSRoutes`?

Reply:
Yup it is to satisfy the API. I think I just didn't add it yet for this PR. Is it easier to merge this with out (assuming the rest is ok), then follow up with a new PR to add that with corresponding tests?

Reply:
OK with me